src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Models/ResourceManagerClient.cs

```diff
@@ -12,22 +12,29 @@
 // limitations under the License.
 // ----------------------------------------------------------------------------------
 
-using System;
-using System.Linq;
-using System.Collections.Generic;
-using Microsoft.Rest.Azure.OData;
-using Microsoft.Azure.Management.ResourceManager;
-using Microsoft.Azure.Management.ResourceManager.Models;
-using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkClient;
+using Microsoft.Azure.Commands.Common.Authentication;
 using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
 using Microsoft.Azure.Management.ContainerRegistry.Models;
+using Microsoft.Azure.Management.ResourceManager;
+using Microsoft.Azure.Management.ResourceManager.Models;
+using Microsoft.Rest.Azure.OData;
+using System;
+using System.Collections.Generic;
+using System.Linq;
 
 namespace Microsoft.Azure.Commands.ContainerRegistry
 {
-    public class ResourceManagerClient : ResourceManagerSdkClient
+    public class ResourceManagerClient
     {
-        public ResourceManagerClient(IAzureContext context) : base(context)
+        private ResourceManagementClient _client;
+
+        public Action<string> VerboseLogger { get; set; }
+        public Action<string> ErrorLogger { get; set; }
+        public Action<string> WarningLogger { get; set; }
+
+        public ResourceManagerClient(IAzureContext context)
         {
+            _client = AzureSession.Instance.ClientFactory.CreateArmClient<ResourceManagementClient>(context, AzureEnvironment.Endpoint.ResourceManager);
```

Comment:
@djyou we should remove all dependencies external to your service where we can, which would include this dependency on the `Microsoft.Azure.Management.ResourceManager` package. Is there any reason that you are using this package instead of the common ARM code we have in [`Commands.ResourceManager.Common`](https://github.com/Azure/azure-powershell/tree/preview/src/ResourceManager/Common/Commands.ResourceManager.Common/Generated)?

Reply:
Are you suggesting we should also remove references such as https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/ContainerRegistry/Commands.ContainerRegistry/Commands.ContainerRegistry.csproj#L54?

Reply:
@djyou correct, your project shouldn't be dependent on anything but SDKs that your service manages and common code projects.